pkg/controllers/resources/nodes/translate.go

```diff
@@ -281,3 +286,36 @@ func mergeResources(a corev1.ResourceList, b corev1.ResourceList) corev1.Resourc
 	}
 	return merged
 }
+
+func (s *nodeSyncer) filterOutTaintsMatchingTolerations(taints []corev1.Taint) []corev1.Taint {
+	var filtered []corev1.Taint
+
+nextTaint:
+	for _, taint := range taints {
+		for _, tol := range s.enforcedTolerations {
+			// Refer Kubernetes Documentation for Taints & Tolerations
+			// for rules about toleration matching taints
+			// https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+
```

Comment:
We don't need to provide official documentation link. As a comment, we only provide the info about what or why we have implemented certain code or if there is any special case like you mentioned below.